products/sentinel_one.py

```diff
@@ -163,6 +115,86 @@ def _authenticate(self):
+
+        # extract account/site ID from configuration if set
+        if 'account_id' in config[self.profile]:
+            for id in config[self.profile]['account_id'].split(','):
+                if id not in account_ids:
+                    account_ids.append(id.strip())
+
+        if 'site_id' in config[self.profile]:
+            for id in config[self.profile]['site_id'].split(','):
+                if id not in site_ids:
+                    site_ids.append(id.strip())
+
+        if 'account_name' in config[self.profile]:
+            for name in config[self.profile]['account_name'].split(','):
+                if name not in account_names:
+                    account_names.append(name.strip())
+
+        # determine site and account IDs to query (default is all)
+        self._site_ids = list()
+        self._account_ids = list()
+
+        if account_ids: # verify provided account IDs are valid
+            response = self._get_all_paginated_data(self._build_url(f'/web/api/v2.1/accounts'),
+                                                    params={'ids': ','.join(account_ids)},
+                                                    add_default_params=False)
+            for account in response:
+                if account['id'] not in self._account_ids:
+                    self._account_ids.append(account['id'])
+
+            diff = list(set(account_ids) - set(self._account_ids))
+            if len(diff) > 0:
+                self.log.warning(f'Account IDs {",".join(diff)} not found.')
+
+        temp_account_name = list()
+        for name in account_names: # verify provided account names are valid
+            response = self._get_all_paginated_data(self._build_url('/web/api/v2.1/accounts'),
+                                                    params={'name': name},
+                                                    add_default_params=False)
+            for account in response:
+                temp_account_name.append(account['name'])
+                if account['id'] not in self._account_ids:
+                    self._account_ids.append(account['id'])
+        
+        diff = list(set(account_names) - set(temp_account_name))
+        if len(diff) > 0:
+            self.log.warning(f'Account names {",".join(diff)} not found')
+
+        
+        if site_ids: # ensure specified site IDs are valid and not already covered by the account_ids listed above
+            temp_site_ids = list()
+            response = self._get_all_paginated_data(self._build_url('/web/api/v2.1/sites'),
+                                                    params={'siteIds': ','.join(site_ids)},
+                                                    add_default_params=False)
+            for item in response:
+                for site in item['sites']:
+                    temp_site_ids.append(site['id'])
+                    if site['accountId'] not in self._account_ids and site['id'] not in self._site_ids:
```

Comment:
```suggestion
                    if site['id'] not in self._site_ids:
```
Do not check if site is in a specified accountID as we are no longer pulling all siteIds for each account so these should be unique sites that allow for filtering specific sites within a specified accountId

Reply:
The logic behind this check is based on the fact site IDs roll up into account IDs.
We've already processed the provided account IDs and they will be used for scoping. If a site ID falls under one of the account IDs provided, we don't need to add that to the scope since it's redundant.

Reply:
This is problematic if I specify an account ID in the config or cmdline and want to only specify a single site ID for a query as it will still query the entire account ID. The suggested change will only add site IDs that are specifically specified so if only account ID is specified it will query the account, while if siteID is specified it acts as a filter to search only that site within the account.

Reply:
I see where you're coming from - but shouldn't providing account_id be avoided (given that it's an optional parameter) if you were wanting to keep it narrowly scoped to a specific site? 

My thoughts initially go to the situation where you want to query an account and also a site outside of that defined account. For example, you want to search all of CompanyA but only the HR site for CompanyB?

Reply:
we have a console where we have 1 account and huge number of sites. we configure the account ID in the config file and then specify the specific site we want to query via commandline. this prevents us from needing to setup a profile for every single new site that is created. If this is not how the tool was designed, maybe this specific change is better fit into a separate feature request. We also have troubles for our onebox where we have a huge number of accounts, and right now the api tries to add them all to the query and errors out due to the query being too long.

Reply:
new commit with the following logic should resolve this issue

```
if site_id in account_id
    remove account_id from scope
    add site_id to stope
else: 
    add site_id to scope
```